modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -247,18 +311,22 @@ public GridEncryptionManager(GridKernalContext ctx) {
 
     /** {@inheritDoc} */
     @Override public void onDisconnected(IgniteFuture<?> reconnectFut) {
-        synchronized (genEcnKeyMux) {
+        synchronized (opsMux) {
             assert !disconnected;
 
             disconnected = true;
 
+            pendingMasterKey = null;
```

Comment:
Why we just nullify variables here?
What will happen if process in progress?
How it would be finished?

Reply:
Reconnecting is identical to the node join procedure. If the client node disconnects from the cluster during the key change process, it should reset these variables. The client node will not be able to join/reconnect during the master key change procedure.

Reply:
Should we complete change master key future?
>     public IgniteFuture<Void> changeMasterKey(String masterKeyName);

Reply:
Yes, it will be completed in `cancelFutures("Client node was disconnected from topology (operation result is unknown).");`